linux/system/repo.sls

```diff
@@ -140,6 +137,24 @@ linux_repo_{{ name }}:
   {%- endif %}
   {%- endif %}
 
+{%- if repo.get('key') %}
+
+linux_repo_{{ name }}_key:
+  cmd.run:
+    - name: "echo '{{ repo.key }}' | apt-key add -"
+    - onchange:
```

Comment:
change this requisite to: require_in [to be processed before the repo]
and add the idemopotency clausule:
unless: "make repo.key fingerpring | apt-key list | grep key fingerprint"

Reply:
actuall syntax is "onchanges" - https://docs.saltstack.com/en/latest/ref/states/requisites.html

@cznewt, I would give a try, it's actually questionable whether on change implicitly require previous state to have a result - so it's actually as require_in. The doc in this case is not clear. Quite sure it would be safer to user
```
require_in: 
  - linux_repo_{{ name }}
```